schema.ddl

```diff
@@ -73,3 +144,92 @@ ON subnet (provider_id);
 
 CREATE UNIQUE INDEX IF NOT EXISTS idx_subnet_id
 ON subnet (id);
+
+/*
+ * Model
+ */
+CREATE TABLE IF NOT EXISTS model (
+	uuid					TEXT PRIMARY KEY,
+	id						TEXT NOT NULL UNIQUE,
+	name					TEXT,
+	type					TEXT,
+	life					INT,
+	owner					TEXT,
+	migration_mode			TEXT,
+	environ_version			INT,
+	last_available_tools	TEXT,
+	meter_status			TEXT,
+	password_hash			TEXT,
+	force_destroyed			TEXT,
+	destroy_timeout			TEXT
+	-- cloud
+	-- cloud_region
+	-- cloud_credential
+	-- controller_id
+	-- sla
+);
+
+CREATE UNIQUE INDEX IF NOT EXISTS idx_model_id
+ON model (id);
+
+/*
+ * Application
+ */
+CREATE TABLE IF NOT EXISTS application (
+	uuid					TEXT PRIMARY KEY,
+	id						TEXT NOT NULL UNIQUE,
+	name					TEXT,
+	series					TEXT,
+	subordinate				BOOLEAN,
+	charm_url				TEXT,
+	channel					TEXT,
+	charm_origin			TEXT,
+	charm_modified_version	INT,
+	force_charm				BOOLEAN,
+	life					INT,
+	min_units				INT,
+	exposed					BOOLEAN,
+	desired_scale			INT,
+	password_hash			TEXT,
+	placement				TEXT,
+	has_resources			BOOL
+	-- unit_count
```

Comment:
Same for these counts. They'll be implied by RI.

Reply:
That's why I left them commented, to remind me an equivalent will need to be implemented at some point

Apparently this didn't occur to me for attachment_count however!